engine/execution/provider/engine.go

```diff
@@ -185,15 +188,36 @@ func (e *Engine) onChunkDataRequest(
 		Collection:    collection,
 	}
 
-	// sends requested chunk data pack to the requester
-	err = e.chunksConduit.Unicast(response, originID)
-	if err != nil {
-		return fmt.Errorf("could not send requested chunk data pack to (%s): %w", origin, err)
+	sinceProcess := time.Since(processStart)
+
+	log = log.With().Dur("sinceProcess", sinceProcess).Logger()
+
+	if sinceProcess > time.Second*10 {
```

Comment:
Not sure how permanent it is, but maybe would be good to make this limit configureable?